Machine-Teaching-Examples/moab_pomdp_esn.ink

```diff
@@ -0,0 +1,146 @@
+###
+
+# MSFT Bonsai 
+# Copyright 2021 Microsoft
+# This code is licensed under MIT license (see LICENSE for details)
+
+# Moab sample illustrating usage of memory, as a Machine Teaching strategy,
+# for problems with missing state information, i.e. partially observable.
+# Using a history of states and actions can help when a sensor cannot read
+# all the relevant information to describe the dynamics.
+
+# We recommend including state information if you can measure it (velocity),
+# but adding memory is helpful when features are hard to define or don't
+# capture everything that is relevant.
+
+# Training and exporting a brain with memory handles the history for you 
+# internally, i.e. you simply need to provide the current ObservableState.
+
+###
+
+inkling "2.0"
+
+using Math
+using Goal
+
+# Distances measured in meters
+const RadiusOfPlate = 0.1125 # m
+
+# Velocities measured in meters per sec.
+const MaxVelocity = 6.0
+const MaxInitialVelocity = 1.0
+
+# Threshold for ball placement
+const CloseEnough = 0.02
+
+# Default time delta between simulation steps (s)
+const DefaultTimeDelta = 0.045
+
+# Maximum distance per step in meters
+const MaxDistancePerStep = DefaultTimeDelta * MaxVelocity
+
+# State received from the simulator after each iteration
+type SimState {
+    # Ball X,Y position
+    ball_x: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+    ball_y: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+
+    # Ball X,Y velocity
+    ball_vel_x: number<-MaxVelocity .. MaxVelocity>,
+    ball_vel_y: number<-MaxVelocity .. MaxVelocity>,
+}
+
+# Imagine there isn't a sensor for measuring velocity and so we only provide
```

Comment:
There isn't a sensor for velocity!

Suggested:
Deliberately remove ball velocity from the state passed to the brain, and give it only the ball position.